src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts

```diff
@@ -161,18 +182,61 @@ declare const acquireVsCodeApi: any;
       line.appendChild(code);
     }
   }
-  function getCurrentSeverity() {
-    const stringMap = {
+
+  /**
+   * Transforms a severity string from a `Suggestion` object into a `CurrentSeverity` object.
+   *
+   * This function maps a severity string ('Low', 'Medium', 'High') to its corresponding
+   * numeric value. If the provided severity string is not one of the allowed values, the function
+   * returns `undefined`.
+   *
+   * @param {Suggestion['severity']} severity - The severity string to be mapped.
+   * @returns {CurrentSeverity | undefined} The mapped severity object, or undefined
+   */
+  function getCurrentSeverity(severity: Suggestion['severity']): CurrentSeverity | undefined {
+    const severityMap = {
       Low: 1,
       Medium: 2,
       High: 3,
     };
-    return suggestion
-      ? {
-          value: stringMap[suggestion.severity],
-          text: suggestion.severity,
-        }
-      : undefined;
+
+    const severityAllowedValues = Object.keys(severityMap);
+    if (!severityAllowedValues.includes(severity)) {
+      return undefined;
+    }
+
+    return {
+      value: severityMap[severity],
+      text: severity,
+    };
+  }
+
+  /**
+   * Toggles visibility of severity icons based on the current severity level.
+   *
+   * If `currentSeverity` is undefined, all icons are hidden.
+   * It also updates the title attribute of the `severity` element to the text
+   * representation of the current severity.
+   *
+   * @param {HTMLElement} severity - The HTML element containing severity icons.
+   * @param {CurrentSeverity | undefined} currentSeverity - The current severity object, or
+   *        undefined if there is no current severity.
+   */
+  function toggleSeverityIcons(severity: HTMLElement, currentSeverity: CurrentSeverity | undefined) {
+    const severityIcons = severity.querySelectorAll('img');
+    const validIconsIds: { [key: number]: string } = { 1: 'sev1', 2: 'sev2', 3: 'sev3' };
+
+    if (!currentSeverity) {
+      severityIcons.forEach(icon => (icon.className = 'icon hidden'));
+      return;
+    }
+
+    severity.setAttribute('title', currentSeverity.text);
+
+    severityIcons.forEach(icon => {
+      const currentIconId = validIconsIds[currentSeverity.value];
+      icon.className = icon.id === currentIconId ? 'icon' : 'icon hidden';
+    });
   }
```

Comment:
I've opted for strict comparison `icon.id === currentIconId` instead of testing it with `contains`: `n.id.includes(currentSeverity.value)` to ensure matching